### PR TITLE
Fix initial disbursement logic

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Fixes how initial disbursements work! Also write tests covering every permutation of permission, over/under default amount, and account approval   

**Todo**

- [ ] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [ ] Request review from relevant @person or team
- [ ] QA your pull request. This includes running the app, login, testing changes etc.
- [ ] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by michielderoos** [comment]
[Review Approve and Disburse](https://trello.com/c/u5pvqAUn/539-review-approve-and-disburse)

File: app/server/api/transfer_account_api.py [hunk 1]
```diff
@@ -114,7 +117,7 @@ def put(self, transfer_account_id):
                 transfer_account.payable_epoch = payable_epoch
 
             if not approve == transfer_account.is_approved and transfer_account.is_approved is not True:
-                transfer_account.approve_and_disburse()
+                transfer_account.approve_initial_disbursement()
```
Comment (enjeyw) [inline app/server/api/transfer_account_api.py:120]: I think this is unreachable due to:
```
     if approve: 
                transfer_account.is_approved = True
```

Shouldn't they together be:

```
if not transfer_account.is_approved and approve:
                transfer_account.is_approved = True
                transfer_account.approve_initial_disbursement()
```

File: app/server/api/transfer_account_api.py [hunk 2]
```diff
@@ -143,9 +146,11 @@ def put(self, transfer_account_id):
                     })
 
                     continue
+                if approve: 
+                    transfer_account.is_approved = True
 
                 if not transfer_account.is_approved and approve:
-                    transfer_account.approve_and_disburse()
+                    transfer_account.approve_initial_disbursement()
```
Comment (enjeyw) [inline app/server/api/transfer_account_api.py:153]: same here

**Review by enjeyw — CHANGES_REQUESTED** [review]
Just something I'm not quite sure about in the top level of the API

**Review by enjeyw — APPROVED** [review]
Looks good!

**Pull request merged**